### PR TITLE
[12.0][IMP] add deprecation warning to `subscription.request.create_comp_sub_req()`

### PR DESCRIPTION
**Pull request by huguesdk**
add deprecation warning to `subscription.request.create_comp_sub_req()`.

related to #390

### CODE REVIEW

File: cooperator/models/subscription_request.py [hunk 1]
```diff
@@ -123,10 +124,13 @@ def create(self, vals):
...
 
-    # todo: remove this and use create() instead. must stay in 12.0 (as
-    # deprecated) but be removed in 14.0.
     @api.model
     def create_comp_sub_req(self, vals):
+        warnings.warn(
+            "subscription.request.create_comp_sub_req() is deprecated. "
+            "please use .create() instead.",
+            DeprecationWarning,
+        )
```
Comment (robinkeunen) [inline cooperator/models/subscription_request.py:133]: c'est un warning pour l'utilisateur ? Je pense qu'un warning dans les logs suffit.

**Review by robinkeunen — CHANGES_REQUESTED** [review]

File: cooperator/models/subscription_request.py (hunk 1 above)
Comment (huguesdk) [reply cooperator/models/subscription_request.py:133]: no, it should only appear as a log message, although i could not yet make it appear. it is used like this in some parts of odoo core as well as some oca modules. doc: https://docs.python.org/3/library/warnings.html#warnings.warn

Comment (robinkeunen) [reply cooperator/models/subscription_request.py:133]: Ok then, TIL => thanks

**Review by robinkeunen — APPROVED** [review]

**Pull request merged**